### ISSUE TITLE
Compare error and metric curves to original training

### ISSUE THREAD

**Issue by vitalwarley** [vitalwarley/research#13]
[Log](https://raw.githubusercontent.com/anxiangsir/insightface_arcface_log/master/ms1mv3_r100_lr02/training.log).

Seems to me that my LR is not decreasing as it should be.

**Comment by vitalwarley** [comment]
After doing #15, let's check the total steps. 

### Insightface

Right log to check is [here](https://raw.githubusercontent.com/anxiangsir/insightface_arcface_log/master/ms1mv2_r100/training.log).

```
...
Training: 2022-04-27 01:27:19,617-: batch_size               128
Training: 2022-04-27 01:27:19,617-: optimizer                sgd
Training: 2022-04-27 01:27:19,617-: lr                       0.1
Training: 2022-04-27 01:27:19,617-: momentum                 0.9
Training: 2022-04-27 01:27:19,617-: weight_decay             0.0005
...
Training: 2022-04-27 01:27:19,617-: num_classes              85742
Training: 2022-04-27 01:27:19,617-: num_image                5822653
Training: 2022-04-27 01:27:19,617-: num_epoch                20
Training: 2022-04-27 01:27:19,617-: warmup_epoch             0
Training: 2022-04-27 01:27:19,617-: val_targets              ['lfw', 'cfp_fp', 'agedb_30']
Training: 2022-04-27 01:27:19,617-: total_batch_size         1024
Training: 2022-04-27 01:27:19,617-: warmup_step              0
Training: 2022-04-27 01:27:19,618-: total_step               113720
```

### Mine

```
(research) 18:23:31 λ  research main ✗ guild run tcc:pretrain-ddp batch-size=1024 accumulate_grad_batches=1 devices=0,1 nprocs=2 max_epochs=20                                                                                                                                         
You are about to run tcc:pretrain-ddp                                                                                                                                                                                                                                                  
  accumulate_grad_batches: 1                                                                                                                                                                                                                                                           
  arcface-m: 0.5                                                                                                                                                                                                                                                                       
  arcface-s: 64                                                                                                                                                                                                                                                                        
  batch-size: 1024                                                                                                                                                                                                                                                                     
  cooldown: 0                                                                                                                                                                                                                                                                          
  data-dir: /home/warley/dev/datasets/MS1M_v2                                                                                                                                                                                                                                          
  devices: 0,1                                                                                                                                                                                                                                                                         
  gradient_clip_val: 5                                                                                                                                                                                                                                                                 
  loss: arcface                                                                                                                                                                                                                                                                        
  lr: 0.1                                                                                                                                                                                                                                                                              
  max_epochs: 20                                                                                                                                                                                                                                                                       
  model: resnet101                                                                                                                                                                                                                                                                     
  momentum: 0.9                                                                                                                                                                                                                                                                        
  nprocs: 2                                                                                                                                                                                                                                                                            
  precision: 16                                                                                                                                                                                                                                                                        
  scheduler: poly                                                                                                                                                                                                                                                                      
  task: pretrain                                                                                                                                                                                                                                                                       
  warmup: 0                                                                                                                                                                                                                                                                            
  weight-decay: 0.0005                                                                                                                                                                                                                                                                 
Continue? (Y/n)  
```

results in 

```
09-Jul-22 18:24:11 - INFO - config:configure_optimizers:182 - Model will train for steps=56880
09-Jul-22 18:24:11 - INFO - config:configure_optimizers:182 - Model will train for steps=56880
```

for each GPU, therefore 113760 steps!

**Comment by vitalwarley** [comment]
The last insightface's logged training loss and LR were 0.4557 and 0.0000, respectively. My loss was a bit higher, close to 1, but the lowest was 0.667.

Training accuracy wasn't computed in insightface, but I dit it and achieved 0.99 from the epoch 10 till 20.

Validation accuracy for insightface were

```
Training: 2022-04-27 13:01:41,424-[lfw][112000]XNorm: 21.779004
Training: 2022-04-27 13:01:41,424-[lfw][112000]Accuracy-Flip: 0.99750+-0.00281
Training: 2022-04-27 13:01:41,425-[lfw][112000]Accuracy-Highest: 0.99817
Training: 2022-04-27 13:02:31,736-[cfp_fp][112000]XNorm: 22.111145
Training: 2022-04-27 13:02:31,737-[cfp_fp][112000]Accuracy-Flip: 0.98529+-0.00542
Training: 2022-04-27 13:02:31,737-[cfp_fp][112000]Accuracy-Highest: 0.98629
Training: 2022-04-27 13:03:15,249-[agedb_30][112000]XNorm: 22.232803
Training: 2022-04-27 13:03:15,250-[agedb_30][112000]Accuracy-Flip: 0.98167+-0.00882
Training: 2022-04-27 13:03:15,250-[agedb_30][112000]Accuracy-Highest: 0.98250
```

and in my case

```
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
     Validate metric           DataLoader 0
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
         lfw/acc            0.9969999194145203
        lfw/norm             21.31365203857422
      lfw/threshold          1.241999864578247
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       Test metric             DataLoader 0             DataLoader 1
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
      agedb_30/acc          0.9699999690055847       0.9699999690055847
      agedb_30/norm          21.14153289794922        21.14153289794922
   agedb_30/threshold       1.2899999618530273       1.2899999618530273
       cfp_fp/acc           0.9494285583496094       0.9494285583496094
       cfp_fp/norm          19.374977111816406       19.374977111816406
    cfp_fp/threshold        1.3299999237060547       1.3299999237060547
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

I didn't achieve the right results for `agedb` and `cfp`, but I improved them with respect to the [last time](https://github.com/vitalwarley/research/issues/9#issuecomment-1120193092)!

**Comment by vitalwarley** [comment]
Unfortunately, my AUC on FIW is a bit off yet: 0.6717 (it should be 0.7183).
